src/distribution/laplace.rs

```diff
@@ -0,0 +1,363 @@
+use distribution::{Continuous, Univariate};
+use rand::distributions::Distribution;
+use rand::distributions::OpenClosed01;
+use rand::Rng;
+use statistics::*;
+use std::f64;
+use {Result, StatsError};
+
+/// Continuous Univariate Laplace distribution.
+/// The Laplace distribution is a distribution over the real numbers parameterized by a mean and
+/// scale parameter. The PDF is:
+///     p(x) = \frac{1}{2 * scale} \exp{- |x - mean| / scale}.
+/// <a href="http://en.wikipedia.org/wiki/Laplace_distribution">Wikipedia - Laplace distribution</a>.
+pub struct Laplace {
+    location: f64,
+    scale: f64,
+}
+
+impl Laplace {
+    fn sample_unchecked<R: Rng + ?Sized>(r: &mut R, location: f64, scale: f64) -> f64 {
+        let r: f64 = r.gen();
+        let u = r - 0.5;
+        location - (scale * f64::signum(u) * f64::ln(1.0 - (2.0 * f64::abs(u))))
+    }
+
+    fn valid_parameter_set(location: f64, scale: f64) -> bool {
+        scale > 0.0 && !location.is_nan()
+    }
+
+    /// Initializes a new instance of the Laplace struct.
+    /// returns an error is scale is negative
+    pub fn new(location: f64, scale: f64) -> Result<Laplace> {
+        if Laplace::valid_parameter_set(location, scale) {
+            Ok(Laplace { location, scale })
+        } else {
+            Err(StatsError::BadParams)
+        }
+    }
+}
+
+impl Mean<f64> for Laplace {
+    /// Gets the mean of the distribution.
```

Comment:
There's a pretty standard template for documentation we've been using of

```
/// Returns the {attribute} of the X distribution
///
/// # Formula
///
/// ```ignore
/// {formula_for_attribute}
///  ```
```

Feel free to reference one of the other distributions